### DISCUSSION TITLE
360 ° flip

### DISCUSSION THREAD

**Discussion by AsiaShi** [bitcraze/discussions#211] [Q&A] [answered]
I want to use Crazyflie do a 360 ° turning maneuver flight, However, the problem is that when the UAV turns 180 °, the motor will stop running and the UAV will enter the stop state. How can I release this constraint?

**Comment by knmcguire** [comment]
Hi! 

That is due to a safety feature calleed the tumble detection: https://github.com/bitcraze/crazyflie-firmware/blob/55d18b0839c03b5194b2574fee89229003b749c0/src/modules/src/supervisor.c#L119-L121. 

Currently we don't have any firmware setting to turn that off so you would need to remove those lines and prevent it from putting it in emergency stop mode. 

There are some things that we should improve for this detector as it is not very clear that it is enabled: https://github.com/bitcraze/crazyflie-firmware/issues/1002

**Comment by AsiaShi** [comment, marked as answer]
Thank you for your reply, I have increased the time to detect the tumble state, about 2s; Therefore, when I perform 360 ° flip operation, the UAV will not stop in an emergency, but when the tumbling time exceeds 2s, an emergency stop will be triggered. This is a better method I can think of at present. I have achieved 360 ° Flip using crazyflie 2.1.
